docs/setup_snapshotter_by_daemonset.md

```diff
@@ -0,0 +1,282 @@
+    # If `daemon_mode` option is not specified, the default value is multiple.
+    daemon_mode = "multiple"
+
+    [daemon]
+    # Specify a configuration file for nydusd
+    nydusd_config = "/opt/nydus/conf/nydusd-fusedev.json"
+    nydusd_path = "/opt/nydus/bin/nydusd"
+    nydusimage_path = "/opt/nydus/bin/nydus-image"
+    # fusedev or fscache
+    fs_driver = "fusedev"
+
+    [log]
+    # Print logs to stdout rather than logging files
+    log_to_stdout = true
+    # Snapshotter's log level
+    level = "info"
+```
+
+### Same filesystem with different snapshotter configuration and different nydusd configuration
+
+If we want to setup the nydus snapshotter for the same fs_driver (such as `fusedev`) with different snapshotter configuration and different nydusd configuration, we can enable `ENABLE_SNAPSHOTTER_CONFIG_FROM_VOLUME` and add the snapshotter configuration (named `config.toml`) in the `Configmap` in `nydus-snapshotter.yaml`:
+
+```yaml
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nydus-snapshotter-configs
+  labels:
+    app: nydus-snapshotter
+  namespace: nydus-snapshotter
+data:
+  ENABLE_SNAPSHOTTER_CONFIG_FROM_VOLUME: "true"
+
+  config.toml: |-
+    version = 1
+    # Snapshotter's own home directory where it stores and creates necessary resources
+    root = "/var/lib/containerd-nydus"
+    # The snapshotter's GRPC server socket, containerd will connect to plugin on this socket
+    address = "/run/containerd-nydus/containerd-nydus-grpc.sock"
+    # The nydus daemon mode can be one of the following options: multiple, dedicated, shared, or none. 
+    # If `daemon_mode` option is not specified, the default value is multiple.
+    daemon_mode = "multiple"
+
+    [daemon]
+    # Specify a configuration file for nydusd
+    nydusd_config = "/opt/nydus/conf/nydusd-config.json"
+    nydusd_path = "/opt/nydus/bin/nydusd"
+    nydusimage_path = "/opt/nydus/bin/nydus-image"
+    # fusedev or fscache
+    fs_driver = "fusedev"
+
+    [log]
+    # Print logs to stdout rather than logging files
+    log_to_stdout = true
+    # Snapshotter's log level
+    level = "info"
+
+  nydusd-config.json: |-
+    {
+      "device": {
+        "backend": {
+          "type": "registry",
+          "config": {
+            "timeout": 5,
+            "connect_timeout": 5,
+            "retry_limit": 2
+          }
+        },
+        "cache": {
+          "type": "blobcache"
+        }
+      },
+      "mode": "direct",
+      "digest_validate": false,
+      "iostats_files": false,
+      "enable_xattr": true,
+      "amplify_io": 1048576,
+      "fs_prefetch": {
+        "enable": true,
+        "threads_count": 8,
+        "merging_size": 1048576,
+        "prefetch_all": true
+      }
+    }
+```
+
+**NOTE:** We need to set `nydusd_config` to `/opt/nydus/conf/nydusd-config.json` in the `config.toml`, so that snapshotter can find the nydusd configuration from configmap.
+
+
+
+### Customized Options
```

Comment:
Not sure about providing these options, it seems like it increases the cost of understanding and uses for the user, can we provide `misc/snapshotter/config.toml` and `misc/snapshotter/nydusd-config-xxx.json` as the only configuration entry points?

Reply:
I agree simplifying the options. We should keep the default configs and allow user-defined configs, but require both snapshotter config and nydusd config for customization.

Reply:
Done.